### PR TITLE
feat(completion): add codeium-cmp plugin

### PR DESCRIPTION
**Pull request by khas-amir**
## 📑 Description

Add native codeium  plugin for neovim

### CODE REVIEW

File: lua/astrocommunity/completion/codeium-cmp/init.lua [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  {
+    "Exafunction/codeium.nvim",
+    config = function()
+      require("codeium").setup {
+        enable_chat = true,
+      }
+    end,
+  },
+  {
+    "hrsh7th/nvim-cmp",
```
Comment (Uzaaft) [inline lua/astrocommunity/completion/codeium-cmp/init.lua:12]:
```suggestion
    "hrsh7th/nvim-cmp",
    optional = true,
```

**Review by Uzaaft — COMMENTED** [review]
We should probably split this up into two folders.

**Comment by khas-amir** [comment]
@Uzaaft  why ?

**Comment by Uzaaft** [comment]
Nevermind, turns out the plugin doesn't provide virtual_text stuff. What a shame.

**Review by Uzaaft — COMMENTED** [review]
Astrocore should be a dependency of the main plugin

File: lua/astrocommunity/completion/codeium-cmp/init.lua [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  {
+    "onsails/lspkind.nvim",
+    optional = true,
+    -- Adds icon for codeium using lspkind
+    opts = function(_, opts) opts.symbol_map = { Codeium = "" } end,
+  },
+  {
+    event = "User AstroFile",
+    dependencies = { "Exafunction/codeium.nvim" },
+    "AstroNvim/astrocore",
```
Comment (Uzaaft) [inline lua/astrocommunity/completion/codeium-cmp/init.lua:31]:
```suggestion
    "AstroNvim/astrocore",
    event = "User AstroFile",
    dependencies = { "Exafunction/codeium.nvim" },

```

**Comment by Uzaaft** [comment]
@khas-amir  We'd appreciate it if you next time follow the conventionsin this repository.

**Review by Uzaaft — DISMISSED** [review]

File: lua/astrocommunity/completion/codeium-nvim/init.lua [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            },
+            ["<Leader>;a"] = {
+              desc = "Auth",
+              function() vim.cmd "Codeium Auth" end,
+            },
+          },
+        },
+      })
+    end,
+  },
```
Comment (Uzaaft) [inline lua/astrocommunity/completion/codeium-nvim/init.lua:66]:
```suggestion
```

File: lua/astrocommunity/completion/codeium-nvim/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# codeium-nvim
+
+A native neovim extension for Codeium
+
+Usage:
+
+- `:Codeium Auth` - to set up the plugin
+- `:Codeium Chat` - Open chat in browser
+
+**Repository**: <https://github.com/Exafunction/codeium.nvim>
```
Comment (Uzaaft) [inline lua/astrocommunity/completion/codeium-nvim/README.md:10]:
```suggestion
**Repository**: <https://github.com/Exafunction/codeium.nvim>


_Note_: You need to auth first before usage
```

**Comment by Uzaaft** [comment]
Removed mapping that didn't make sense at all. Should be good to go now

**Review by Uzaaft — DISMISSED** [review]

**Comment by Uzaaft** [comment]
Just tested before I merged this, and seems like the cmp lazy loading isn't working 🫠

**Comment by Uzaaft** [comment]
Should be working now. @khas-amir  Can you test before I merge this?

**Comment by khas-amir** [comment]
@Uzaaft  everything works on my arch linux

**Comment by khas-amir** [comment]
community.lua 
```lua
return {
  { "AstroNvim/astrocommunity", dev = true },
  { import = "astrocommunity.colorscheme.catppuccin" },
  { import = "astrocommunity.motion.nvim-surround" },
  { import = "astrocommunity.pack.lua" },
  { import = "astrocommunity.pack.rust" },
  { import = "astrocommunity.pack.typescript" },
  { import = "astrocommunity.pack.tailwindcss" },
  { import = "astrocommunity.completion.codeium-nvim" },
}

``` `

**Review by Uzaaft — APPROVED** [review]

**Pull request merged**